lib/main.js

```diff
@@ -1,18 +1,37 @@
 const WritableStream = require('stream').Writable
 const inherits = require('util').inherits
 
+const MultipartParser = require('./types/multipart')
+const UrlencodedParser = require('./types/urlencoded')
 const parseParams = require('./utils').parseParams
 
 function Busboy (opts) {
   if (!(this instanceof Busboy)) { return new Busboy(opts) }
-  if (opts.highWaterMark !== undefined) { WritableStream.call(this, { autoDestroy: false, highWaterMark: opts.highWaterMark }) } else { WritableStream.call(this, { autoDestroy: false }) }
+
+  if (typeof opts !== 'object') {
+    throw new TypeError('Busboy expected an options-Object.')
+  }
+  if (typeof opts.headers !== 'object') {
+    throw new TypeError('Busboy expected an options-Object with headers-attribute.')
+  }
+  if (typeof opts.headers['content-type'] !== 'string') {
+    throw new TypeError('Missing Content-Type-header.')
+  }
+
+  const {
+    headers,
+    ...streamOptions
+  } = opts
+
+  this.opts = {
+    autoDestroy: false,
+    ...streamOptions
+  }
+  WritableStream.call(this, this.opts)
 
   this._done = false
-  this._parser = undefined
+  this._parser = this.parseHeaders(headers)
```

Comment:
@kibertoad 

Should I rename parseHeaders to getParserByHeaders?

Reply:
if this is internal only, sure

Reply:
done :)